src/averageAges.js

```diff
@@ -55,7 +56,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? child.sex === 'm' && people.some(mother => mother.name === child.mother)
+    : people.some(mother => mother.name === child.mother));
+  const averageAge = children.reduce((sum, child) => {
+    const mother = people.find(woman => woman.name === child.mother);
+
+    return sum + ((child.born - mother.born));
+  }, 0) / children.length;
+
+  return averageAge;
+}
+
+function calcAverAge(objects) {
```

Comment:
```suggestion
function calculateAverageAge(people) {
```
It is better to use full names for functions and meaningful names for parameters